src/vs/base/common/event.ts

```diff
@@ -165,7 +165,10 @@ export namespace Event {
 	export function any<T>(...events: Event<T>[]): Event<T>;
 	export function any(...events: Event<any>[]): Event<void>;
 	export function any<T>(...events: Event<T>[]): Event<T> {
-		return (listener, thisArgs = null, disposables?) => combinedDisposable(...events.map(event => event(e => listener.call(thisArgs, e), null, disposables)));
+		return (listener, thisArgs = null, disposables?) => {
+			const disposable = combinedDisposable(...events.map(event => event(e => listener.call(thisArgs, e))));
```

Comment:
There was previously a bug here, in that the disposiable created from `combinedDisposables` always 'wanted' to be disposed, but if a DisposableStore was passed to the event then it was unnecessary